### PR TITLE
Add Invitations API methods

### PR DESCRIPTION
**Pull request by LauraBeatris**
## Description

Adds `get_invitation`, `list_invitation`, `send_invitation` and `revoke_invitation` methods

## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[X] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.

### CODE REVIEW

File: lib/workos/user_management.rb [hunk 1]
```diff
@@ -678,6 +678,118 @@ def send_password_reset_email(email:, password_reset_url:)
...
 
+      # Gets an Invitation
+      #
+      # @param [String] id The unique ID of the Invitation.
+      #
+      # @return WorkOS::Invitation
+      sig do
+        params(id: String).returns(WorkOS::Invitation)
+      end
+      def get_invitation(id:)
```
Comment (LauraBeatris) [inline lib/workos/user_management.rb:689]: All of those methods should be added to the [API reference](https://workos.com/docs/reference/user-management/password-reset/complete) as well.

File: lib/workos.rb [hunk 2]
```diff
@@ -43,37 +43,37 @@ def self.key
     config.key
   end
 
-  autoload :Types, 'workos/types'
```
Comment (LauraBeatris) [inline lib/workos.rb:46]: Fixing the alphabetical order here.

File: lib/workos/user_management.rb [hunk 3]
```diff
@@ -678,6 +678,118 @@ def send_password_reset_email(email:, password_reset_url:)
...
+          request: get_request(
+            path: '/user_management/invitations',
+            auth: true,
+            params: options,
+          ),
+        )
+
+        parsed_response = JSON.parse(response.body)
+
+        invitations = parsed_response['data'].map do |user|
```
[the remark this answers is not in the document]
Comment (LauraBeatris) [reply lib/workos/user_management.rb:729]: Same https://github.com/workos/workos-ruby/pull/242#discussion_r1406704551

File: spec/support/fixtures/vcr_cassettes/user_management/list_invitations/with_after.yml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      X-Request-Id:
+      - f0ca1391-24f7-42ab-96e1-03708170a415
+      Etag:
+      - W/"59a-ZIKgWSqL79Vk1PhmBPwwz8b4bhU"
+      Via:
+      - 1.1 vegur
+      Cf-Cache-Status:
+      - DYNAMIC
+      Report-To:
+      - '{"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v3?s=zcH%2FppoPRB%2BmUlJabP9ZSUe5cPHU7%2FBBbFhBBbfnttHbFCsiSIpwTqTp4tefv%2BR6SEeQEpNhai3Ra7Zs7YmwYxdAw0ECz4bFh9D1NJeV%2BBEvWpoaIj475X5J%2FiPdUkqGAhTQ9nKkIeMW4MJIkw%3D%3D"}],"group":"cf-nel","max_age":604800}'
```
[the remark this answers is not in the document]
Comment (LauraBeatris) [reply spec/support/fixtures/vcr_cassettes/user_management/list_invitations/with_after.yml:68]: To be fair, I'm not really aware of where this comes from, but I can see that this is being applied in a lot of fixtures so I presume that it isn't a problem.

File: lib/workos/invitation.rb [hunk 5]
```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+# typed: true
+
+module WorkOS
+  # The Invitation class provide a lightweight wrapper around a WorkOS Invitation
```
Comment (jbvolta) [inline lib/workos/invitation.rb:5]: provides

**Review by jbvolta — APPROVED** [review]

File: lib/workos/invitation.rb (hunk 5 above)
Comment (LauraBeatris) [reply lib/workos/invitation.rb]: Fixed on https://github.com/workos/workos-ruby/pull/242/commits/0a9b66f13ce2407d9a9e955e42fb9ad509dd854c

**Pull request merged**